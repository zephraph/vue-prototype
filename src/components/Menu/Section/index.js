export { default } from './Section.vue';
