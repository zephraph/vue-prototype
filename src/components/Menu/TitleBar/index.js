export { default } from './TitleBar.vue';
