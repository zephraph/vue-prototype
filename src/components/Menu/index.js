import MenuSection from './Section';
import MenuField from './Field';

export default {
  functional: true,
  render: (h, { props }) =>
    <MenuSection title={ props.title }></MenuSection>,
  props: {
    title: String,
    fields: Array
  },
  components: {
    MenuField
  }
}
