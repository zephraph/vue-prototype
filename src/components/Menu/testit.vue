<template lang="pug">
  .TestIt
</template>
