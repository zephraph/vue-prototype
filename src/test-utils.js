import { createRenderer } from 'vue-server-renderer';
import Vue from 'vue';

export const render = async (component) => {
  const renderer = createRenderer();
  return new Promise((resolve, reject) => {
    renderer.renderToString(new Vue(component), (err, str) => {
      if (err) reject(err);
      resolve(str);
    });
  });
}
