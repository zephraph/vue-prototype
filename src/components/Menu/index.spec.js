import Menu from './index';
import { render } from 'test-utils';

const makeProps = props => ({
  'props': {
    ...props
  }
});

test('Should create an empty menu with the given title', async () => {
  const props = makeProps({
    title: 'test'
  });
  const result = await render(h => <Menu try="this" {...props}/>);

  expect(result).toMatchSnapshot();
});
