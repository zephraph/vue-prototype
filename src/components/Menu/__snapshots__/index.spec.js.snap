exports[`test Should create an empty menu with the given title 1`] = `
<div server-rendered="true" class="Menu-Section">
    <div class="Menu-TitleBar"><span>test</span></div>
</div>
`;
